packages/calypso-e2e/src/lib/flows/difm-flow.ts

```diff
@@ -0,0 +1,89 @@
+import { Page } from 'playwright';
+import { DataHelper } from '../..';
+
+const selectors = {
+	// General
+	button: ( text: string ) => `button:text("${ text }")`,
+
+	// Start page
+	startContainer: '.signup.is-do-it-for-me',
+
+	// Options page
+	siteName: '#siteTitle',
+	tagline: '#tagline',
+
+	// Social page
+	facebook: 'input[name=facebookUrl]',
+};
+
+/**
+ * Class encapsulating the flow when starting a new do-it-for-me site order.
+ */
+export class DIFMFlow {
+	/**
+	 * Constructs an instance of the flow.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( private page: Page ) {}
+
+	/**
+	 * Given text, click on the button's first instance with the text.
+	 *
+	 * @param {string} text User-visible text on the button.
+	 */
+	async clickButton( text: string ): Promise< void > {
+		const selector = selectors.button( text );
+
+		await this.page.waitForSelector( selector );
+		await this.page.click( selector );
+	}
+
+	/**
+	 * Go to first setup page.
+	 */
+	async startSetup(): Promise< void > {
+		await this.page.goto(
+			DataHelper.getCalypsoURL(
+				'/start/do-it-for-me/new-or-existing-site?flags=signup/redesigned-difm-flow'
+			)
+		);
+		await this.validateSetupPage();
+		await this.clickButton( 'Start a new site' );
+	}
+
+	/**
+	 * Validates that we've landed on the setup page.
+	 */
+	async validateSetupPage(): Promise< void > {
+		await this.page.waitForSelector( selectors.startContainer );
+	}
+
+	/**
+	 * Validates that we've landed on the options page.
+	 */
+	async validateOptionsPage(): Promise< void > {
+		await this.page.waitForSelector( selectors.siteName );
+	}
+
+	/**
+	 * Validates that we've landed on the social page.
+	 */
+	async validateSocialPage(): Promise< void > {
+		await this.page.waitForSelector( selectors.facebook );
+	}
+
+	/**
+	 * Enter the values for your site name as well as an optional tagline.
+	 *
+	 * @param {string} name The name for the new site.
+	 * @param {string} [tagline] An optional tagline.
+	 */
+	async enterOptions( name: string, tagline?: string ): Promise< void > {
+		await this.page.fill( selectors.siteName, name );
+		if ( typeof tagline !== 'undefined' ) {
+			await this.page.fill( selectors.tagline, tagline );
+		}
+		await this.clickButton( 'Continue' );
```

Comment:
I would also do a similar thing here where I pull the button click out of this method and into its own action in the script.

The reason for that is twofold:
1. It makes this method a little more reusable in the future. For example, maybe someone wants to enter values into these fields, but then go backward in the flow, and see if state is preserved when they return.
2. It makes the testing script read as a little more explicit. 
For example, in the script, this currently ends up looking like...
```typescript
await difmFlow.enterOptions( 'Test Site', 'Just a little test' );
await difmFlow.validateSocialPage();
```
In reading that at a surface level, it's not clear how we got from entering the site options to the new social page.

Let me know if that doesn't make sense!

Reply:
Yep. This would have limited us in future PRs!